pkg/services/ngalert/schedule/fetcher.go

```diff
@@ -50,7 +50,5 @@ func (sch *schedule) updateSchedulableAlertRules(ctx context.Context, disabledOr
 		return fmt.Errorf("failed to get alert rules: %w", err)
 	}
 	sch.schedulableAlertRules.set(q.Result)
-	sch.metrics.SchedulableAlertRules.Set(float64(len(q.Result)))
```

Comment:
I think I want to run with that risk and not have it here too.

Reply:
What I mean is we can keep it here and then remove it from `schedulePeriodic`, since `schedulePeriodic` cannot schedule rules without first calling `updateSchedulableAlertRules`?

Reply:
There is a difference - The actual set of alerts used comes from `schedulableAlertRules.all()` which could have changed between the `sch.schedulableAlertRules.set` and the call to `sch.schedulableAlertRules.all`